### PR TITLE
make virtual service tests workload agnostic

### PR DESCRIPTION
**Pull request by stevenctl**
- [x] use templated config
- [x] remove Call/Target/Count
- [x] trim down the # of sources/destinations
- [x] debug duplicate top level tests (virtualservice/a#01/...)

### CODE REVIEW

File: tests/integration/pilot/common/routing.go [hunk 1]
```diff
@@ -79,52 +79,48 @@ spec:
...
+			name: "added header",
+			config: `
 apiVersion: networking.istio.io/v1alpha3
 kind: VirtualService
 metadata:
   name: default
 spec:
   hosts:
-  - b
+  - {{ (index .dst 0).Config.Service }}
```
Comment (howardjohn) [inline tests/integration/pilot/common/routing.go:94]: nit: is it possible to add a function on `echo.Instances` for .Service()? Then we can access it by `dst.Service`

Comment (howardjohn) [reply tests/integration/pilot/common/routing.go:94]: since this is super common

File: tests/integration/pilot/common/traffic.go [hunk 2]
```diff
@@ -62,8 +62,19 @@ type TrafficTestCase struct {
...
+
+	// sourceFilters allows adding additional filtering for workload agnostic cases to test using fewer clients
+	sourceFilters []echotest.SimpleFilter
+	// targetFilters allows adding additional filtering for workload agnostic cases to test using fewer targets
+	targetFilters []echotest.SimpleFilter
 }
 
+var (
+	// TODO move to echotest?
+	defaulltSourceFilters = []echotest.SimpleFilter{echotest.SingleSimplePodBasedService, echotest.NoExternalServices}
```
Comment (howardjohn) [inline tests/integration/pilot/common/traffic.go:74]: nit: typo in defaullt

**Review by howardjohn — APPROVED** [review]

**Comment by JimmyCYJ** [comment]
@stevenctl could you please cherry-pick to release-1.9?

**Comment by stevenctl** [comment]
I'll restore cherry-pick when the dependency cherr-pick lands: #31498

**Pull request merged**